Source/WebCore/page/PerformanceUserTiming.cpp

```diff
@@ -100,7 +100,7 @@ ExceptionOr<Ref<PerformanceMark>> PerformanceUserTiming::mark(JSC::JSGlobalObjec
     if (markOptions && markOptions->startTime)
         timestamp = m_performance.monotonicTimeFromRelativeTime(*markOptions->startTime);
 
-    InspectorInstrumentation::performanceMark(context.get(), markName, timestamp, is<Document>(context) ? downcast<Document>(context).frame() : nullptr);
+    InspectorInstrumentation::performanceMark(context.get(), markName, timestamp, is<Document>(context) ? downcast<Document>(context.get()).frame() : nullptr);
```

Comment:
Just picking this as an example, there's other examples of this pattern in the patch.

Are `foo.get()` (or `foo.ptr()`) necessary? I think if you just pass `context` here you'll get the const ref overload of downcast which will call `get` for you, right?

It would be a nice API for downcast if we don't have to think about getting the raw pointer or raw reference out of the smart pointer when downcasting which I think is how it would work as written.

Reply:
`downcast<Document>(context).frame()` wouldn't build because `context` is a smart pointer and so downcast would return a smart pointer to. I could have used `downcast<Document>(context)->frame()` to make it build but that would create a little bit of unnecessary ref-counting churn. As a result, I decided to go with `downcast<Document>(context.get()).frame()` to avoid the churn (minor optimization).

Reply:
Or were you talking about the first parameter (`context.get()`) which I didn't change in my patch?

Reply:
I was talking about the downcast parameter, I should have picked a better example where there wasn't an ambiguity.

That makes sense, you specifically wanted the raw reference in this case and likely in all the other cases where you use `.get` when calling `downcast`.